api.R

```diff
@@ -1,101 +1,232 @@
 # Setup ------------------------------------------------------------------------
 library(arrow)
+library(assertthat)
 library(aws.s3)
 library(ccao)
 library(dplyr)
 library(lightsnip)
+library(tibble)
 library(plumber)
+library(purrr)
+library(rapidoc)
 library(vetiver)
 source("generics.R")
 
+# Define constants
+dvc_bucket_pre_2024 <- "s3://ccao-data-dvc-us-east-1"
+dvc_bucket_post_2024 <- "s3://ccao-data-dvc-us-east-1/files/md5"
+base_url_prefix <- "/predict"
+
 # Read AWS creds from Docker secrets
 if (file.exists("/run/secrets/ENV_FILE")) {
   readRenviron("/run/secrets/ENV_FILE")
-} else {
+} else if (file.exists("secrets/ENV_FILE")) {
   readRenviron("secrets/ENV_FILE")
 }
 readRenviron(".env")
 
 # Get the model run attributes at runtime from env vars
-dvc_bucket <- Sys.getenv("AWS_S3_DVC_BUCKET")
 run_bucket <- Sys.getenv("AWS_S3_MODEL_BUCKET")
-run_id <- Sys.getenv("AWS_S3_MODEL_RUN_ID")
-run_year <- Sys.getenv("AWS_S3_MODEL_YEAR")
 api_port <- as.numeric(Sys.getenv("API_PORT", unset = "3636"))
-
-
-# Download Files ---------------------------------------------------------------
-
-# Grab model fit and recipe objects
-temp_file_fit <- tempfile(fileext = ".zip")
-aws.s3::save_object(
-  object = file.path(
-    run_bucket, "workflow/fit",
-    paste0("year=", run_year),
-    paste0(run_id, ".zip")
+default_run_id_var_name <- "AWS_S3_DEFAULT_MODEL_RUN_ID"
+default_run_id <- Sys.getenv(default_run_id_var_name)
+
+# The list of runs that will be deployed as possible model endpoints
+valid_runs <- rbind(
```

Comment:
**suggestion (non-blocking):** It might be worth just updating the `model.final_model` dbt seed to capture this list and its attributes. That seed only gets updated at the end of each year, so we could just basically down > up the compose stack to get the new endpoint, rather than modifying the code.

If this sounds like a good idea, let's push it to another separate PR.

Reply:
Good call, issue opened here: https://github.com/ccao-data/data-architecture/issues/428